### PR TITLE
Issue 21796 custom content tool endpoint

### LINKED ISSUE #21796: Create and endpoint to add a custom content tool
**Is your feature request related to a problem? Please describe.**
As a developer, I want to be able to **create and add to the menu** a custom content tool to the menu from an endpoint.

**Describe the solution you'd like**
The endpoint should receive:

1. Name of the content tool
2. The content type
3. If default view is card or list
4. Menu under where to be added

For the menu, I think we can use the `id`, that way in the UI we can use the menu endpoint (`/api/v1/menu`) to create a selector to pick where to put it.

**Additional context**
Just for an idea, [this is the UI](https://user-images.githubusercontent.com/751424/156364426-9e2656b5-4ef9-4647-861b-8f7966bd0abe.png) we use now.

But after this, then we have to added under a menu, the idea of this endpoint is to do both tasks at the same time.

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**
This endpoint brings the possibility of adding a custom content-tool portlet into a layout

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/portlet/PortletResource.java [hunk 1]
```diff
@@ -109,6 +109,74 @@ public final Response createContentPortlet(@Context final HttpServletRequest req
...
+
+            final List<String> portletIds = new ArrayList<>(layout.getPortletIds());
+            portletIds.add(portlet.getPortletId());
+            layoutAPI.setPortletIdsToLayout(layout, portletIds);
+
+            return Response.ok(new ResponseEntityView(
+                    map("portlet", portlet.getPortletId(), "layout", layout.getId())))
+                    .build();
+
+        } catch (Exception e) {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/portlet/PortletResource.java:172]: you can remove the catch, Jersey handles the exceptions for us

**Review by jdotcms — APPROVED** [review]

**Review by freddyucv — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/portlet/PortletResource.java (hunk 1 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/portlet/PortletResource.java]: done

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/portlet/PortletResource.java [hunk 2]
```diff
@@ -109,6 +111,75 @@ public final Response createContentPortlet(@Context final HttpServletRequest req
...
+    public final Response addContentPortletToLayout(@Context final HttpServletRequest request,
+            @PathParam("portletId") final String portletId,
+            @PathParam("layoutId") final String layoutId) throws DotDataException {
+
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requiredBackendUser(true)
+                .requiredFrontendUser(false)
+                .requestAndResponse(request, null)
+                .rejectWhenNoUser(true)
+                .requiredPortlet("roles")
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/portlet/PortletResource.java:133]: This should be a constant on Portlet

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/portlet/PortletResource.java [hunk 3]
```diff
@@ -109,6 +111,75 @@ public final Response createContentPortlet(@Context final HttpServletRequest req
...
+        if (!portletAPI.canAddPortletToLayout(portletId)) {
+            return ResponseUtil.INSTANCE
+                    .getErrorResponse(request, Response.Status.UNAUTHORIZED, user.getLocale(),
+                            user.getUserId(), "custom.content.portlet.add.restricted", portletId);
+        }
+
+        final Portlet portlet = portletAPI.findPortlet(portletId);
+        if (null == portlet) {
+
+           final String errorMessage = Try.of(()->LanguageUtil.get( request.getLocale(), "custom.content.portlet.not.found", portletId ))
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/portlet/PortletResource.java:150]: shouldn't be the user locale?

**Review by jdotcms — APPROVED** [review]

**Review by freddyucv — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

**Pull request merged**